stdlib/ballerina-http/src/main/ballerina/http/http_retry_client.bal

```diff
@@ -35,224 +36,243 @@ public type RetryClient object {
 
-        P{{path}} Target service url.
-        P{{request}} A request message.
+        P{{path}} Resource path
+        P{{request}} An HTTP outbound request message
+        R{{}} The HTTP `Response` message, or an error if the invocation fails
     }
-    public function delete (string path, Request? request = ()) returns Response|HttpConnectorError;
+    public function delete(string path, Request? request = ()) returns Response|HttpConnectorError;
 
     documentation {
-        The GET function implementation of the HTTP retry client. Protects the invocation of the GET function
-        attached to the underlying HTTP client.
+        The `get()` function wraps the underlying HTTP actions in a way to provide
+        retrying functionality for a given endpoint to recover from network level failures.
 
-        P{{path}} Target service url.
-        P{{request}} A request message.
+        P{{path}} Resource path
+        P{{request}} An HTTP outbound request message
+        R{{}} The HTTP `Response` message, or an error if the invocation fails
     }
-    public function get (string path, Request? request = ()) returns Response|HttpConnectorError;
+    public function get(string path, Request? request = ()) returns Response|HttpConnectorError;
 
     documentation {
-        The OPTIONS function implementation of the HTTP retry client. Protects the invocation of the OPTIONS function
-        attached to the underlying HTTP client.
+        The `options()` function wraps the underlying HTTP actions in a way to provide
+        retrying functionality for a given endpoint to recover from network level failures.
 
-        P{{path}} Target service url.
-        P{{request}} A request message.
+        P{{path}} Resource path
+        P{{request}} An HTTP outbound request message
+        R{{}} The HTTP `Response` message, or an error if the invocation fails
     }
-    public function options (string path, Request? request = ()) returns Response|HttpConnectorError;
+    public function options(string path, Request? request = ()) returns Response|HttpConnectorError;
 
     documentation {
-        The SUBMIT function implementation of the HTTP retry client.
+	Submits an HTTP request to a service with the specified HTTP verb.
+	The `submit()` function does not give out a `Response` as the result,
+	rather it returns an `HttpFuture` which can be used to do further interactions with the endpoint.
 
-        P{{httpVerb}} HTTP verb to be used for the request.
-        P{{path}} Target service url.
-        P{{request}} A request message.
-    }
-    public function submit (string httpVerb, string path, Request request) returns HttpFuture|HttpConnectorError;
+        P{{httpVerb}} The HTTP verb value
+        P{{path}} The resource path
+        P{{request}} An HTTP outbound request message
+        R{{}} An `HttpFuture` that represents an asynchronous service invocation, or an error if the submission fails
+	}
```

Comment:
formatting issue? with all the other functions in this obbject after this point too